src/solvers/flattening/boolbv.cpp

```diff
@@ -674,24 +674,16 @@ bool boolbvt::is_unbounded_array(const typet &type) const
 {
   if(type.id()==ID_symbol)
     return is_unbounded_array(ns.follow(type));
-
   if(type.id()!=ID_array)
     return false;
-
   if(unbounded_array==unbounded_arrayt::U_ALL)
     return true;
 
   const exprt &size=to_array_type(type).size();
-
-  mp_integer s;
-  if(to_integer(size, s))
-    return true;
-
-  if(unbounded_array==unbounded_arrayt::U_AUTO)
-    if(s>MAX_FLATTENED_ARRAY_SIZE)
-      return true;
-
-  return false;
+  if(auto s = numeric_cast<mp_integer>(size))
+    return unbounded_array == unbounded_arrayt::U_AUTO &&
+           *s > MAX_FLATTENED_ARRAY_SIZE;
+  return true;
```

Comment:
To be honest, I don't see how this is a "simplification". I'd much more accept that if you really gathered all the bits together. How about this:

````c++
if(type.id()==ID_symbol)
  return is_unbounded_array(ns.follow(type));

const exprt &size=to_array_type(type).size();
return
  type.id() == ID_array &&
  (unbounded_array == unbounded_arrayt::U_ALL ||
   !(auto s = numeric_cast<mp_integer>(size)) ||
   (unbounded_array == unbounded_arrayt::U_AUTO &&
    *s > MAX_FLATTENED_ARRAY_SIZE));
````

Reply:
return `(auto s = func()) && s > 0)`

Reply:
Ok, thanks, so just declare `s` beforehand and just do the assignment in there.

Reply:
Yeah; not sure this change is actually a simplification.  Also -- what problem is this solving?

Reply:
Changed according to Michael's comment.